ouroboros-network/src/Ouroboros/Network/TxSubmission/Inbound.hs

```diff
@@ -350,12 +360,21 @@ txSubmissionInbound _tracer maxUnacked mpReader mpWriter _version =
                               + fromIntegral (Seq.length acknowledgedTxIds)
         }
       where
+
+        -- TODO: Add to StrictSeq.
+        partition :: (a -> Bool) -> StrictSeq a -> (StrictSeq a, StrictSeq a)
+        partition cond = foldl' part (Seq.empty, Seq.empty)
+          where
+            part (ls, rs) e
+              | cond e    = ((Seq.|>) ls e, rs)
+              | otherwise = (ls, (Seq.|>) rs e)
```

Comment:
Alternatively:
```suggestion
        partition :: (a -> Bool) -> StrictSeq a -> (StrictSeq a, StrictSeq a)
        partition cond xs = (Seq.toStrict ls, Seq.toStrict rs)
          where
            (ls, rs) = LazySeq.partition cond xs
```
with: `import qualified Data.Sequence as LazySeq`
I'm not sure which one is more efficient.